src/main/java/com/maxmind/db/Metadata.java

```diff
@@ -32,18 +27,37 @@
 
     private final int searchTreeSize;
 
-    Metadata(JsonNode metadata) {
-        this.binaryFormatMajorVersion = metadata.get(
-                "binary_format_major_version").asInt();
-        this.binaryFormatMinorVersion = metadata.get(
-                "binary_format_minor_version").asInt();
-        this.buildEpoch = metadata.get("build_epoch").asLong();
-        this.databaseType = metadata.get("database_type").asText();
-        this.languages = metadata.get("languages");
-        this.description = metadata.get("description");
-        this.ipVersion = metadata.get("ip_version").asInt();
-        this.nodeCount = metadata.get("node_count").asInt();
-        this.recordSize = metadata.get("record_size").asInt();
+    @MaxMindDbConstructor
+    public Metadata(
+            @MaxMindDbParameter(name="binary_format_major_version")
+            int binaryFormatMajorVersion,
+            @MaxMindDbParameter(name="binary_format_minor_version")
+            int binaryFormatMinorVersion,
+            @MaxMindDbParameter(name="build_epoch")
+            BigInteger buildEpoch,
+            @MaxMindDbParameter(name="database_type")
+            String databaseType,
+            @MaxMindDbParameter(name="languages")
+            List<String> languages,
+            @MaxMindDbParameter(name="description")
+            Map<String, String> description,
```

Comment:
Ah, `Map<String, String>` does work. I suppose we could be stuffing anything into the `Map` without type safety due to type erasure though, right?

Reply:
Yeah I think that is accurate.

Reply:
Presumably if we work out how to use `getActualTypeArguments`, we can do this in a type-safe way.